### PR TITLE
Introduced relations in OT

### LINKED ISSUE #1468: Finish relations mechanism in OT
Relations mechanism that brings better contextual information for OT and is used by undo has not been finished in the initial refactor PR. This issue is set as a reminder to introduce this mechanism afterward.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Introduced relations in OT which helps solving conflicts during undo. Closes #1468.

### CODE REVIEW

**Comment by scofalik** [comment]
This PR reduces the number of skipped tests to 4, of which only 2 are in transformations (the other 2 are related to a different part of the changed code - engine debug tools).

**Comment by scofalik** [comment]
After some changes, it requires https://github.com/ckeditor/ckeditor5-engine/pull/1486

File: tests/model/operation/transform.js [hunk 1]
```diff
@@ -53,6 +53,12 @@ describe( 'transform', () => {
 		}
 	}
 
+	const contextIsStrong = {
```
Comment (ma2ciek) [inline tests/model/operation/transform.js]: IMO the name suggests a boolean option, not an object.

File: src/model/operation/transform.js [hunk 2]
```diff
@@ -187,9 +203,160 @@ function initializeContext( opsA, opsB, options ) {
 		return this.document.history.isUndoneOperation( originalOp );
 	};
 
+	context.getRelation = function( opA, opB ) {
+		if ( !options.useContext ) {
+			return null;
+		}
+
+		const origB = this.originalOperations.get( opB );
+		const undoneB = this.document.history.getUndoneOperation( origB );
```
Comment (pjasiun) [inline src/model/operation/transform.js]: Undone operation? Comment is needed.

Comment (scofalik) [reply src/model/operation/transform.js]: I am not sure what exactly should be commented here, but I'll add extensive comments in a followup to explain every of those algorithms.

Comment (pjasiun) [reply src/model/operation/transform.js]: Nevermind, it was late. ;)

**Comment by pjasiun** [comment]
Moved to https://github.com/ckeditor/ckeditor5-engine/pull/1497.

**Pull request closed without merging**